src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,126 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K,V>[] table;
+    private int capacity;
+    private int threshold;
+    private int size;
 
     @Override
     public void put(K key, V value) {
-
+        if (table == null) {
+            resize();
+        }
+        int bucket = getBucket(key);
+        Node<K, V> current = table[bucket];
+        if (current != null) {
+            while (current.next != null) {
+                if (isEqualsKeys(current.key, key)) {
+                    current.value = value;
+                    return;
+                }
+                current = current.next;
+            }
+            if (isEqualsKeys(current.key, key)) {
+                current.value = value;
+                return;
+            }
+            if (size == threshold) {
+                resize();
+                bucket = getBucket(key);
+                current = table[bucket];
+                if (current != null) {
+                    while (current.next != null) {
+                        current = current.next;
+                    }
+                    current.next = new Node<>(key, value, null);
+                } else {
+                    current.next = new Node<>(key, value, null);
+                }
+            } else {
+                current.next = new Node<>(key, value, null);
+            }
+        } else {
+            if (size == threshold) {
+                resize();
```

Comment:
too complex logic, simplify it. An algorithm should be like this:
1. check size, if size == threshold, resize array
2. Find the first node in a bucket by key.
3. In case a bucket is null just set a new node for it
4. In case a bucket isn't empty you should go through a chain of nodes using the condition while(node != null)
5. In the case keys equal change value for this key and break loop
6. In the case of node.next == null set a new node as next for node and break loop